### ISSUE TITLE
Hard (sometimes impossible) to use Wekan on mobile (Android & iOS) because of UI/UX issues

### ISSUE THREAD

**Issue by neooleg** [wekan/wekan#953]
## Disclaimer
Apologize for putting a set of issues inside one thread, but as for me this is a single issue with break down which is **almost killing any usage on Wekan on mobile devices**. In the same time, mobile is crucial since today [it's more popular than desktop among users](http://www.smartinsights.com/mobile-marketing/mobile-marketing-analytics/mobile-marketing-statistics/).

*However, if it's not acceptable (put all issues inside) -- just ping me, I'll separate them*.

## Environment
The following issues were found during working with Wekan (different environments, e.g. [Road Map board](https://wekan.indie.host/b/t2YaGmyXgNkppcFBq/wekan-fork-roadmap) or Sandstorm) on mobile both Android and iOS (see versions below)

## Bugs
iOS 10.2.1 (Safari)
- [ ] ❗️ Double tap required to: add card, edit new card title, close card adding, closing card (the most annoying bugs)
- [ ] Useless jumping scroll effect if top/bottom reached. IMO the effect need to be prohibited.
- [ ] No app icon if add on SpringBoard (it seems there are missing/incorrect [touch icons](https://mathiasbynens.be/notes/touch-icons)).

Android 5.0 (Chrome 56)
- [ ] ❗️Not responding or extremely slow react on tap for: *opening card, closing card*. Although, other action are good, e.g. toggle side panel (very fast) or adding new card (less fast, but acceptable).
- [x] ❗️Impossible to scroll horizontally anywhere (except catching top part of board). Might be related/duplicate #882.

**Comment by xet7** [comment]
Improving viewing attachments on mobile at #201

**Comment by neuroradiology** [comment]
I just realized that one huge issue on mobile browsers (android FF and chrome at least) is their desire to fade out the scroll bars.

in fact, you CAN scroll horizontally, but you have to get your swipe on exactly the right part of the page.

(For people trying to replicate this: hit the < button that makes the members/labels/activity menu open; then hit > to make it disappear _while you are watching the bottom 20% of the browser window_. The horizontal bar will briefly appear there, and if you drag that part of the screen, then it will scroll.)

a couple options: 
* ok: make the horizantal scroll bar visible at all times
* better: propagate all swipe/drag events on a list outside of a card to the scrollbar

**Comment by memeplex** [comment]
I would add here that the triple top bar (fir example, chrome, then sandstorm, then wekan) makes the app barely usable in a small screen.

**Comment by xet7** [comment]
@memeplex 

On mobile Chrome, removing browser bar is not possible.

On mobile Firefox, browser bar etc can be removed with addon:
https://addons.mozilla.org/en-US/android/addon/fullscreen-mobile/

1) Install addon from Firefox Addons menu
2) On Firefox menu at bottom, click "Full screen"
3) When you want to get browser from Full screen, hold long finger on screen on Wekan empty area, and click "Exit Full Screen".

On my Android phone it looks like this:

![mobile-firefox](https://user-images.githubusercontent.com/15545/31561465-a384ad1c-b060-11e7-99df-6604421b5e6b.jpg)

**Comment by lichtamberg** [comment]
> Impossible to scroll horizontally anywhere (except catching top part of board). Might be related/duplicate #882.

This is currently absolutely preventing to work with Wekan on mobile phones here (Android 7.1.1 with Chrome), because most time when we try to swipe/scoll we are moving the columns instead of scrolling - so the only thing it does is to break the arrangement of our wekan board. 

Scrolling by swiping the grey area in the lower area doesn't do anything and can't be used to swipe/scroll.

We are using our wekan board without sandstorm. 
(thanks for this great software though ❤️ !)

**Comment by xet7** [comment]
Please test after I have released Wekan v0.60 (or newer) how this has been improved by  @floatinghotpot and @nztqa , for me it seems that back arrow does work on desktop but I had some problems clicking it on Android.

**Comment by xet7** [comment]
Moved from https://github.com/wekan/wekan/pull/458 to here:

From @lichtamberg

I have following 3 issues on my phone (Samsung Galaxy Note 8 with latest Chrome):

1) its really hard to click on the row names (board titles), most of the time it doesn't work and no click/touch event happens (independent if I click on the title or the arrow on the right). It also doesn't work with the Galaxy S Pen, so it's not related about my fingers :) 
2) Sometimes it starts the swap-rows feature, even if my click was just a "press" - not a drag and drop (maybe it's better to remove/deactivate the swap card-titles feature in general on phones?) 
3) When I achieve to reach a cards page, when I press the back button - I don't go to the cards index page back, but to the general board selection page.

**Comment by biji** [comment]
Using new mobile UI, how to move card between list?

**Comment by xet7** [comment]
@biji 

1) Click card
2) On right side of card name, click 3 lines "Hamburger" menu
3) Scroll down and click "Move Card" or "Copy Card"
4) (Scroll up if needed and) Select to what board and list you want to copy or move card

You can copy or move card to list on the same or another board.

**Comment by shred** [comment]
> Not responding or extremely slow react on tap for: opening card, closing card. Although, other action are good, e.g. toggle side panel (very fast) or adding new card (less fast, but acceptable).

Any progress on this one? On Android/Chrome, I have to tap on a card like a maniac, to eventually open it.

**Comment by xet7** [comment]
@shred 

Mobile Web improvements are planned for sometime 2018:
https://blog.wekan.team/2017/12/2-years-of-wekan-what-we-archieved/index.html

**Comment by xet7** [comment]
## Update 2018-12-26

Currently mobile mode works mostly.

Missing are these:
- Move home button / avatar bar from bottom to top. So at top first is home button / avatar, then others.
- When clicing Move Card, go to correct page position. Currently it's at empty page position, and there is need to scroll page up to see Move Card options. It should work similarly like Copy Card, that is visible.
- Also check that other buttons go to visible page.
- Create toggle/switch to top of page that toggles/switches between mobile and full screen mode. This is needed, because keeping full screen mode does not work in all mobile browsers.
- Try to get Swimlane mode working better on mobile mode.

**Comment by xet7** [comment]
## Addition: Moving board position in Wekan

### Currently
- when clicking card it starts dragging card right away
- to move to other list empty area needs to be carefully clicked and use it to move

### Better would be
- when clicking card it starts to move whole wekan page, like in Trello
- when holding on top of card for a longer time, it starts to drag card.

**Comment by hupptechnologies** [comment]
Working on it.

**Comment by RavetcoFX** [comment]
On Chrome, would this be possible by making it a Progressive Web App?

**Comment by xet7** [comment]
@RavetcoFX 

What be possible? Please give more details.

**Comment by RavetcoFX** [comment]
> @RavetcoFX
> 
> What be possible? Please give more details.

Full screen without the browsing bar

**Comment by xet7** [comment]
Moved to #2879
